File: quests/chapter9/python/devops.py
"""Chapter 9: Dungeon of DevOps - Python implementation"""

from typing import Dict, List, Any


def create_dockerfile(base_image: str, workdir: str, port: int) -> str:
    """
    Create a Dockerfile content string

    Args:
        base_image: Base Docker image (e.g., "node:18-alpine")
        workdir: Working directory in container
        port: Port to expose

    Returns:
        Dockerfile content as multi-line string
    """
    # TODO: Return Dockerfile content as multi-line string
    # Should include FROM, WORKDIR, EXPOSE, and CMD directives
    return ""


def create_docker_compose(service_name: str, image: str, port: str) -> str:
    """
    Create docker-compose.yml content string

    Args:
        service_name: Name of the service
        image: Docker image to use
        port: Port mapping (e.g., "3000:3000")

    Returns:
        docker-compose.yml content as multi-line string
    """
    # TODO: Return docker-compose.yml content as multi-line string
    # Should include version, services, and port mapping
    return ""


def create_health_check(endpoint: str, interval: int) -> Dict[str, Any]:
    """
    Create a health check configuration

    Args:
        endpoint: Health check endpoint
        interval: Check interval in seconds

    Returns:
        Health check configuration dictionary
    """
    # TODO: Return health check configuration object
    # Should have properties: test, interval, timeout, retries
    return {}

File: quests/chapter9/typescript/devops.ts
// Chapter 9: Dungeon of DevOps
// Player solutions will live here.
// TODO: Implement Docker and container orchestration configurations.

export function createDockerfile(baseImage: string, workdir: string, port: number): string {
  // TODO: Return Dockerfile content as multi-line string
  // Should include FROM, WORKDIR, EXPOSE, and CMD directives
  // Example base image: "node:18-alpine", workdir: "/app", port: 3000
  return "";
}

export function createDockerCompose(serviceName: string, image: string, port: string): string {
  // TODO: Return docker-compose.yml content as multi-line string
  // Should include version, services, and port mapping
  // Format: "serviceName" -> { image, ports }
  return "";
}

export function createHealthCheck(endpoint: string, interval: number): object {
  // TODO: Return health check configuration object
  // Should have properties: test, interval, timeout, retries
  // Example: endpoint="/health", interval=30
  return {};
}
